core/src/main/java/org/ehcache/core/internal/service/ServiceLocator.java

```diff
@@ -423,6 +425,23 @@ ServiceMap lookupDependenciesOf(ServiceMap resolved, Class<? extends Service> re
         }
       }
     }
+    final OptionalServiceDependencies optionaAnnotation = clazz.getAnnotation(OptionalServiceDependencies.class);
+    if (optionaAnnotation != null) {
+      for (String className : optionaAnnotation.value()) {
+        try {
+          Class<?> dependency = ClassLoading.getDefaultClassLoader().loadClass(className);
```

Comment:
Hum... For me, optional means "I will use the implementation of this service interface if available and otherwise I will do without". But here, we are requesting a concrete implementation. Which is strange.

Reply:
It is loading the interface defined as an optional dependency: not the impl:

```
@ServiceDependencies({CacheManagerProviderService.class, StatisticsService.class, TimeSourceService.class, ExecutionService.class})
@OptionalServiceDependencies({
  "org.ehcache.clustered.client.service.EntityService",
  "org.ehcache.clustered.client.service.ClusteringService"})
public class DefaultManagementRegistryService extends DefaultManagementRegistry implements ManagementRegistryService, CacheManagerListener {
```